x/upgrade/test/integration_test.go

```diff
@@ -0,0 +1,202 @@
+package test
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/testutil/testnode"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	v1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	v1beta1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/config"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+)
+
+func TestUpgrade(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping SDK integration test in short mode.")
+	}
+	suite.Run(t, new(UpgradeTestSuite))
+}
```

Comment:
This integration test suite is probably overkill, but at least we have a very very high degree of confidence that submitting governance proposals to upgrade won't work.

there might be another test we could add where we manually force an upgrade into the state and check that it doesn't get picked up by begin block, but I opted to avoid that for now. If we decide that its worth it, I'm happy to add it!